### PR TITLE
Add basic Chevrotain tokens/rules/actions for the eligibility parser

### PR DESCRIPTION
**Pull request by mscharfstein**
Works on #163 

Code uses an existing Calculator Chevrotain example as the basis: https://github.com/SAP/chevrotain/blob/master/examples/grammars/calculator/calculator_pure_grammar.js

There are many TODOs left to handle a lot of other operators/math expressions but these are the basic ones that existed in the example and will serve as good reference as we continue to build

### CODE REVIEW

File: client/src/serviceCalculator/actions.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
+/** Code adapted from Chevrotain calculator example:
```
Comment (marcja) [inline client/src/serviceCalculator/actions.js:1]: Move all these files to `client/src/services/formulas`. Note that "services" is used in the software architecture sense, not the social program sense.

File: client/src/serviceCalculator/actions.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    let result = this.visit(ctx.lhs);
+
+    // "rhs" key may be undefined as the grammar defines it as optional (MANY === zero or more).
+    if (ctx.rhs) {
+      ctx.rhs.forEach((rhsOperand, idx) => {
+        // there will be one operator for each rhs operand
+        let rhsValue = this.visit(rhsOperand);
+        let operator = ctx.MultiplicationOperator[idx];
+
+        if (tokenMatcher(operator, Multi)) {
```
Comment (marcja) [inline client/src/serviceCalculator/actions.js:58]: Note that this code is virtually identical to the code above, except for this one line, one word.

File: client/src/serviceCalculator/actions.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  }
+
+  atomicExpression(ctx) {
+    if (ctx.parenthesisExpression) {
+      return this.visit(ctx.parenthesisExpression);
+    } else if (ctx.NumberLiteral) {
+      return parseInt(ctx.NumberLiteral[0].image, 10);
+    } else if (ctx.powerFunction) {
+      return this.visit(ctx.powerFunction);
+    }
```
Comment (marcja) [inline client/src/serviceCalculator/actions.js:76]: TODO: This also where you would need to add support for variables, I think.

File: client/src/serviceCalculator/actions.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  // (<), LTE (<=), NE (<>))
+  // Add action for logical expression (handle ND, FALSE, IF, IFS, NOT, OR,
+  // TRUE, XOR)
+  // Add action for Identifier (set variable like HOUSEHOLD_SIZE to its value)
+}
+
+// We only need a single interpreter instance because our interpreter has no state.
+const interpreter = new CalculatorInterpreter();
+
+const calc = (text) => {
```
Comment (marcja) [inline client/src/serviceCalculator/actions.js:102]: s/calc/evaluate/g

File: client/src/serviceCalculator/calculator.test.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import calc from "./actions";
+
+describe("Calculations", () => {
+  it("can calculate an expression", () => {
+    expect(calc("1 + 2").value).toBe(3);
```
Comment (marcja) [inline client/src/serviceCalculator/calculator.test.js:5]: Note that our formulas are expected to start with an equals sign, just like all spreadsheet formulas. So, "=1+2".

Comment (marcja) [inline client/src/serviceCalculator/calculator.test.js:5]: Add test cases to assert that whitespace doesn't matter, that "=1 + 2" is same as "=1+2 ".

File: client/src/serviceCalculator/lexer.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+});
+const Multi = createToken({
+  name: "Multi",
+  pattern: /\*/,
+  categories: MultiplicationOperator,
+});
+const Div = createToken({
+  name: "Div",
+  pattern: /\//,
+  categories: MultiplicationOperator,
```
Comment (marcja) [inline client/src/serviceCalculator/lexer.js:44]: I'm guessing that Div and Minus aren't really implemented fully...

File: client/src/serviceCalculator/actions.js (hunk 1 above)
Comment (mscharfstein) [reply client/src/serviceCalculator/actions.js:1]: done

File: client/src/serviceCalculator/actions.js (hunk 2 above)
Comment (mscharfstein) [reply client/src/serviceCalculator/actions.js:58]: this line and the one above (ctx.MultiplicationOperator) and also below when we use *= and /= vs += and -=

File: client/src/serviceCalculator/actions.js (hunk 3 above)
Comment (mscharfstein) [reply client/src/serviceCalculator/actions.js:76]: ack

File: client/src/serviceCalculator/calculator.test.js (hunk 5 above)
Comment (mscharfstein) [reply client/src/serviceCalculator/calculator.test.js:5]: hmm didn't realize that. we will need to differentiate from a starting = and a = to indicate a boolean expression (a = b). added a TODO since we'll need to add additional tokens and parser functions

Comment (mscharfstein) [reply client/src/serviceCalculator/calculator.test.js:5]: done

File: client/src/serviceCalculator/lexer.js (hunk 6 above)
Comment (mscharfstein) [reply client/src/serviceCalculator/lexer.js:44]: how come?

File: client/src/services/formulas/calculator.test.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import calc from "./actions";
```
Comment (marcja) [inline client/src/services/formulas/calculator.test.js:1]: s/calc/evaluate/g

Also rename file to evaluator.test.js

File: client/src/services/formulas/actions.js [hunk 8]
```diff
@@ -0,0 +1,120 @@
+/** Code adapted from Chevrotain calculator example:
+ * https://github.com/SAPchevrotain/tree/master/examples/grammars/calculator
```
Comment (tariqajyusuf) [inline client/src/services/formulas/actions.js:2]: Missing a slash:

https://github.com/SAP/chevrotain/tree/master/examples/grammars/calculator

Comment (mscharfstein) [reply client/src/services/formulas/actions.js:2]: done

File: client/src/services/formulas/calculator.test.js (hunk 7 above)
Comment (mscharfstein) [reply client/src/services/formulas/calculator.test.js:1]: done

File: client/src/serviceCalculator/lexer.js (hunk 6 above)
Comment (marcja) [reply client/src/serviceCalculator/lexer.js]: My bad. I read the code more closely after reading your comment. Sorry!
